src/granite/querying.cr

```diff
@@ -73,28 +73,24 @@ module Granite::Querying
     first(clause, params) || raise NotFound.new("Couldn't find " + {{@type.name.stringify}} + " with first(#{clause})")
   end
 
-  # find returns the row with the primary key specified.
-  # it checks by primary by default, but one can pass
-  # another field for comparison
+  # find returns the row with the primary key specified. Otherwise nil.
   def find(value)
-    return find_by(@@primary_name, value)
+    first("WHERE #{@@primary_name} = ?", value)
   end
 
+  # find returns the row with the primary key specified. Otherwise raises an exception.
   def find!(value)
-    return find_by!(@@primary_name, value)
+    find(value) || raise Granite::Querying::NotFound.new("Couldn't find #{{{@type.name.stringify}}} with #{@@primary_name} = #{value}")
```

Comment:
This error message when rendered will read: `Couldn't find User with id = 24`. Could we make that one of:

`Couldn't find a(n) User with id = 24`
`No User found where id = 24`
`No User found with id = 24`

Similarly, for the messages generated find_by! and AssociationCollection#find_by!